File: food-backend/src/models/order.model.ts
import { Schema, model, Document, Types } from "mongoose";
import { Orders } from "../types/order.d";
import { Users } from "../types/user.d";
import { Restaurants } from "../types/restaurant.d";
import { MenuItemType } from "../types/menu.d";

const orderSchema = new Schema<Orders>(
  {
    userId: { type: Schema.Types.ObjectId, ref: "User", required: true },
    restaurantId: { type: Schema.Types.ObjectId, ref: "Restaurant", required: true },
    items: [
      {
        menuItemId: { type: Schema.Types.ObjectId, ref: "MenuItem", required: true },
        quantity: { type: Number, required: true, min: 1 },
      },
    ],
    totalPrice: { type: Number, required: true, min: 0 },
    status: {
      type: String,
      enum: ["pending", "preparing", "on_the_way", "delivered", "cancelled"],
      default: "pending",
    },
    paymentStatus: {
      type: String,
      enum: ["paid", "unpaid"],
      default: "unpaid",
    },
  },
  { timestamps: true }
);

export const Order = model<Orders>("Order", orderSchema);

File: lib/models/order_model.dart
import 'menu_item.dart';
class Order {
  final String id;
  final String userId;
  final String restaurantId;
  final List<OrderItem> items;
  final double totalPrice;
  final String status;
  final String paymentStatus;

  Order({
    required this.id,
    required this.userId,
    required this.restaurantId,
    required this.items,
    required this.totalPrice,
    required this.status,
    required this.paymentStatus,
  });

  factory Order.fromJson(Map<String, dynamic> json) => Order(
        id: json['_id'],
        userId: json['userId'],
        restaurantId: json['restaurantId'],
        items: (json['items'] as List)
            .map((item) => OrderItem.fromJson(item))
            .toList(),
        totalPrice: json['totalPrice'].toDouble(),
        status: json['status'],
        paymentStatus: json['paymentStatus'],
      );
}

class OrderItem {
  final MenuItem menuItemId;
  final int quantity;

  OrderItem({
    required this.menuItemId,
    required this.quantity,
  });

  factory OrderItem.fromJson(Map<String, dynamic> json) => OrderItem(
        menuItemId: MenuItem.fromJson(json['menuItemId']),
        quantity: json['quantity'],
      );
}